### ISSUE TITLE
implicit_integer_linter return object is inconsistent with other linters for NULL lint

### ISSUE THREAD

**Issue by MichaelChirico** [r-lib/lintr#1160]
```
writeLines("1L", tmp <- tempfile())
expr <- lintr::get_source_expressions(tmp)
lintr::implicit_integer_linter()(expr$expressions[[1L]])
# list(NULL)
```

whereas most linters would just return `lint()`.

`flatten_lints()` handles this so it's not really distinguishable in regular usage, but it's making the test in #1153 a bit more difficult because `length(list(NULL))` is 1 whereas `length(list())` is 0.

**Comment by MichaelChirico** [comment]
Other affected linters:

 - `make_linter_from_regex()` --> `no_tab_linter()`
 - `spaces_inside_linter()`
 - `undesirable_operator_linter()`

**Comment by AshesITR** [comment]
The `lapply(ids_with_token...)` idiom could be replaced by `compact(lapply(ids_with_token...))` to fix this, right?

**Comment by MichaelChirico** [comment]
yes, but otoh I thought it might make sense to take the chance to refactor these linters while we're at it
